### PR TITLE
Revscript MonsterType Spell Fixes

### PR DESCRIPTION
**Pull request by theshibbies**
### Pull Request Prelude

- [X] I have followed [proper The Forgotten Server code styling][code].
- [X] I have read and understood the [contribution guidelines][cont] before making this PR.
- [X] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

**Protocol version**
Any

### Changes Proposed
Spells aren't handled correctly in register_monster_type.lua, Not exactly sure if this is the fully correct way to handle these, but at least I am bringing the issue to everyones attention.

**Issues addressed:** 
Within revscript monstertype lua files, spells named "condition" wouldn't be registered and would cause errors. Spells named "drunk" would create condition twice, once at line 281 and again at line 331, which would cause all sorts of crashes. You can reproduce these errors by adding a monster lua file and adding these types of spells below.

`{name ="condition", interval = 2000, chance = 30, type = CONDITION_POISON, startDamage = 0, tick = 4000, minDamage = -440, maxDamage = -520, radius = 5, target = false, effect = CONST_ME_POISONAREA}`

`{name ="drunk", interval = 2000, chance = 10, drunkenness = 25, duration = 5000, range = 7, shootEffect = CONST_ANI_ENERGY}`

The easiest way is to convert the current monster xml files into revscript using this tool:
https://otland.net/threads/ot-monster-converter.277546/

**Note:** You may need to create multiple files containing drunk spells in order to make the crash happen, not sure why though. Sometimes it crashes, sometimes it doesn't, the more files the easier it is to replicate.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by EPuncker** [comment]
great! I would like to hear @soul4soul input on this as well

**Comment by emil92b** [comment]
https://github.com/otland/forgottenserver/blob/master/data/scripts/lib/register_monster_type.lua#L337-L354
`{name ="condition", interval = x, chance = x, condition = { type = x , startDamage = x, minDamage = x, maxDamage = x, duration = x, interval = x}`

**Comment by theshibbies** [comment]
Pretty sure you'll still get an error from here.
https://github.com/otland/forgottenserver/blob/b73ac8c77bd81471eef587f9d0e7c3b83256895c/data/scripts/lib/register_monster_type.lua#L281

**Comment by emil92b** [comment]
https://github.com/otland/forgottenserver/pull/3499
Allow conditions to be used with all predefine creature abilities. Conditions can be used with "melee", "combat", "speed", "condition", "outfit", "invisible", "*field", etc.. Conditions won't work with named spells such as "energy strike".

**Comment by theshibbies** [comment]
I see. Well either way it still causes an error.
image

**Comment by theshibbies** [comment]
> #3499 Allow conditions to be used with all predefine creature abilities. Conditions can be used with "melee", "combat", "speed", "condition", "outfit", "invisible", "*field", etc.. Conditions won't work with named spells such as "energy strike".

Alright, great. I guess @soul4soul will have to check into adapting his system for this format. I reverted the changes for that part. However, the drunk spells are still an issue.

File: data/monster/lua/#example.lua [hunk 1]
```diff
@@ -56,18 +56,22 @@ monster.loot = {
...
+	{name = "combat", type = COMBAT_LIFEDRAIN, chance = 10, interval = 2*1000, length = 8, spread = 0, minDamage = -300, maxDamage = -490, effect = CONST_ME_PURPLEENERGY},
 	{name = "speed", chance = 15, interval = 2*1000, speed = -700, radius = 1, target = true, duration = 30*1000, effect = CONST_ME_MAGIC_RED},
 	{name = "firefield", chance = 10, interval = 2*1000, range = 7, radius = 1, target = true, shootEffect = CONST_ANI_FIRE},
-	{name = "combat", type = COMBAT_LIFEDRAIN, chance = 10, interval = 2*1000, length = 8, spread = 0, minDamage = -300, maxDamage = -490, effect = CONST_ME_PURPLEENERGY}
+	{name = "condition", chance = 1000, range = 1, condition = { type = CONDITION_ENERGY, minDamage = 10, maxDamage = 20, duration = 5000, interval = 1000}},
+	{name = "drunk", chance = 10, interval = 2000, drunkenness = 25, duration = 5000, range = 7, shootEffect = CONST_ANI_ENERGY},
+	{name =	"outfit", chance = 10, interval = 2000, item = 7172, duration = 8000, range = 7, effect = CONST_ME_MAGIC_BLUE}
```
Comment (soul4soul) [inline data/monster/lua/#example.lua:66]: Maybe leave this one off until the outfit attack is fixed?  #3884

**Review by soul4soul — COMMENTED** [review]
Yeah #3499 has the best examples, for real tibia spells and conditions only these two formats would be used

This gives melee with a condition
`{name = "melee", attack = 10, skill = 100, condition = { type = CONDITION_FIRE, minDamage = 50, maxDamage = 200, duration = 5000, interval = 1000}},`

This is an attack that causes a condition only.
`{name = "condition", chance = 1000, range = 1, condition = { type = CONDITION_ENERGY, minDamage = 10, maxDamage = 20, duration = 5000, interval = 1000}},`

Looking at the source code of the converter it looks correct but I'll actually test it when I get a chance.

**Review by soul4soul — APPROVED** [review]

**Comment by soul4soul** [comment]
This could use a rebase and add back the outfits spells and it would be good to go.

**Review by ranisalt — DISMISSED** [review]
Rebased against master

File: data/scripts/lib/register_monster_type.lua [hunk 2]
```diff
@@ -331,12 +331,9 @@ local function AbilityTableToSpell(ability)
 			if outfit then
 				spell:setOutfit(outfit)
 			end
-			if ability.name == "drunk" then
-				spell:setConditionType(CONDITION_DRUNK)
-				if ability.drunkenness then
-					spell:setConditionDrunkenness(ability.drunkenness)
-				end
-			end
+      if ability.drunkenness then
```
Comment (EPuncker) [inline data/scripts/lib/register_monster_type.lua:334]: indentation

**Review by EPuncker — APPROVED** [review]

**Pull request merged**